### ISSUE TITLE
Copy to clipboard function not working

### ISSUE THREAD

**Issue by geoffroyjabouley** [philroberts/FPDB-for-OSX#5]
From the Hand viewer tab, the copy to clipboard function is not working.
I'm using fpdb rc6 on windows 7.

When using fpbd 0.40.5, the following is copied on the clipboard, which can then be easily converted with online tools to then posted on poker forums:

```
Table 'Ribeauvillé I, 478968521' 10-max Seat #3 is the button
Seat 1: stuone ($2.48 in chips) 
Seat 3: pokerfab34 ($1.46 in chips) 
Seat 5: dudesque ($2.02 in chips) 
Seat 6: pedross ($2.32 in chips) 
Seat 8: tycoke ($1.54 in chips) 
Seat 10: TaylorSwift ($2.01 in chips) 
dudesque: posts small blind €0.01
pedross: posts big blind €0.02
*** HOLE CARDS ***
tycoke: calls €0.02
TaylorSwift: raises €0.06 to €0.08
stuone: folds 
pokerfab34: folds 
dudesque: raises €0.18 to €0.26
pedross: calls €0.24
tycoke: calls €0.24
TaylorSwift: calls €0.18
*** FLOP *** [Kc 6c 9d]
dudesque: bets €0.62
pedross: folds 
tycoke: raises €0.66 to €1.28 and is all-in
TaylorSwift: folds 
dudesque: calls €0.66
*** TURN *** [Kc 6c 9d] [9c]
*** RIVER *** [Kc 6c 9d 9c] [8s]
dudesque collected €3.36 from pot
*** SUMMARY ***
Total pot €3.60 | Rake €0.24
Board [Kc 6c 9d 9c 8s]
Seat 1: stuone folded
Seat 3: pokerfab34 folded
Seat 5: dudesque collected (€3.36)
Seat 6: pedross folded
Seat 8: tycoke mucked [Kh Jc] 
Seat 10: TaylorSwift folded
```

Also, could it be possible to add to this function:
- all the involved player hands (on fpdb 0.40.5, my hand is never present (from the example, i add AsAh))
- a "copy to clipboard with stats" function. For example, it would add basic stats (vpip/pfr/agg_factor/hands) after each player name.

```
Stats: VPIP/PFR/AGG/HANDS
Seat 1: stuone ($2.48 in chips) - 20.5/5.6/2.1/256
```

I can help fixing/implementing this functionnality with some support (i know python but not qt)

**Comment by philroberts** [comment]
Works for me on both OSX and Win7.  Right-click a hand, copy to clipboard, open notepad, ctrl-v.  Can you try it with a few different hands to see if some of them work?  Is there anything in the log files that might help track this down?

Thanks for pointing out the lack of hole cards, I just copied the old hand exporting code and didn't look too closely.  That's definitely something we should fix.

As for including stats, that's something I've had as a kind of bluesky idea on my todo list for a while.  The concerns would be format, which stats to include, and the most straightforward way to do it would give you stats as of copy time, not stats as of hand time.  So if you had a hand at the start of a session when you only had 5 hands on your villain, that wouldn't be reflected by the time you copy it at the end of session.

**Comment by geoffroyjabouley** [comment]
Thanks for your anwer. Just for the record, i am the guy you helped fixing display issues with Mucked flop & wrong seat stats display for ipoker sites.

There is indeed an error in the log file, which i can copy to the clipboard ;)

```
   fpdb starting ...Traceback (most recent call last):
   File "Z:\fpdb\build\fpdb\out00-PYZ.pyz\GuiHandViewer", line 254, in copyHandToClipboard
   File "Z:\fpdb\build\fpdb\out00-PYZ.pyz\Hand", line 1176, in writeHand
   File "Z:\fpdb\build\fpdb\out00-PYZ.pyz\Hand", line 1054, in writeHand
   UnicodeEncodeError: 'ascii' codec can't encode character u'\u20ac' in position 48: ordinal not in range(128)
   Traceback (most recent call last):
   File "Z:\fpdb\build\fpdb\out00-PYZ.pyz\GuiHandViewer", line 254, in copyHandToCl
```

I don't know why it is using a path like "Z:\fpdb". I have a network drive labelled Z: , but it is definitely not a drive fpdb should use...

Btw, i am using windows 7 64bit with French locale.

Regarding the stats to include in a copied hand, maybe we can start another thread about this topic.
I will ask on a poker forum what stats and format is used by PokerTracker and HoldemManager softwares.

**Comment by geoffroyjabouley** [comment]
Maybe this could help:
https://stackoverflow.com/questions/9942594/unicodeencodeerror-ascii-codec-cant-encode-character-u-xa0-in-position-20

**Comment by philroberts** [comment]
Ah, python and unicode again.  As a first guess, one of the players in the hand has a screen name with a non-ascii character in it?  I'll try to find a repro case in my db and get on fixing it.  Edit: I think it's worse than that, some currency symbols are non-ascii, and I think that's what is happening to you - '\u20ac' is the euro symbol.

As for the Z:\ path names, that's an artifact of the exe building process - I build from a Z: drive on windows.  My paths get preserved in python's error traces, for reasons I can't fathom and don't want to.

**Comment by philroberts** [comment]
Just fixed it, was a one-liner: a6adedf.  Will be in next release, whenever that is.

**Comment by geoffroyjabouley** [comment]
Thanks for the quick fix regarding € symbol.
I will test it on the next release.
